Goldstone wiki content

HarshaF1 opened the pull request:
I have come up with the basic/initial content for goldstone wiki documentation. For now I have created a .md file in docs folder for discussion/review, once we converge on the content I will copy paste this content to wiki.
I couldn't think of anyother way to get the wiki reviewed. Please let me know if there is a better way to do this.

Sections `Configuration Guide` & `CLI reference guide` are still WIP. 
Meanwhile I would request you to go through the document and provide inputs on
- Add/Modify the content wherever required.
- some content for arm64 platforms and gearbox.

KazuyaAnazawa commented:
@HarshaF1 Thank you so much for your work.
I will give comments after I check the contents in detail.

KazuyaAnazawa commented on docs/wiki_review.md:
```diff
@@ -50,4 +53,495 @@ The Switch Abstraction Interface defines the API to provide a vendor-independent
+| 2    | Wistron  | WTP-01-C1-00 (Galileo Flex T) | Broadcom    | TH+         |                    |                                    |
+| 3    | Edgecore | AS7716-24SC/XC  (Cassini)     | Broadcom    | TH+         | 16x100G + 8x200G   | only ACO PIU is supported          |
+| 4    | Edgecore | AS7716-32X                    | Broadcom    | TH+         | 32x100G            |                                    |
+
```
Could you add the following two platforms?
- Edgecore, AS7316-26XB (CSR320), Broadcom, Qumran-AX, 2x100G + 8x25G + 16x10G
- Edgecore, AS7946-30XB (AGR400), Broadcom, Qumran2C, 4 x 400G + 22 x 100G + 4 x 25G

KazuyaAnazawa commented:
@HarshaF1 I took a look at the documents, and added several comments.
Please take a look. If needed, I can work together.

HarshaF1 commented on docs/wiki_review.md:
```diff
@@ -0,0 +1,53 @@
+
+Several open-source components developed as part of [OCP(Open Compute Project)](https://www.opencompute.org/projects/networking), [TIP(Telcom Infra project)](https://telecominfraproject.com/oopt/) and [ONL(Open Network Linux)](http://opennetlinux.org/) are used as the basic building blocks of Goldstone NOS.
+1. ONL &emsp; &ensp; &ensp; &nbsp;: ONL is used as base operating system, which has support for wide range of open network devices.
+2. SAI &emsp; &ensp; &ensp; &ensp; : Abstraction layer to control/Manage forwarding ASICs/NPUs. 
```
Latest SAI can support DNX (qumran) family of chips from broadcom (https://github.com/sonic-net/sonic-buildimage/blob/2afd3829ad1debf6f63572e61ea420e16e709909/platform/broadcom/sai.mk#L15). Since we are taking about open source components here, I will capture OcNOS information in the architecture section below. 
Please let me if that is okay.

HarshaF1 commented on docs/wiki_review.md:
```diff
@@ -0,0 +1,53 @@
+
+This modular architecture brings in the ease of assimilating new technologies and feature-advancements of open-source projects mentioned above into Goldstone-NOS. And also enables Goldstone to extend its support for other networking devices [L0/L1 Transponders, ROADMs.. etc.] in future.
+
+### _Unified Management Layer_
```
I agree with you and have changed it to 'Goldstone management layer' and also changed the architecture diagram.
I have provided the link to [goldstone-mgmt](https://github.com/oopt-goldstone/goldstone-mgmt) in the description below. Please let me know if you want me to elaborate more on gs-mgmt here in this section.

HarshaF1 commented:
Apologies for the delay in addressing the review comments. 
I have addressed the review comments , could you please take another look. And also please feel free to add more information if required.

ishidawataru commented:
@KazuyaAnazawa Can we merge this PR?

KazuyaAnazawa commented:
@ishidawataru @HarshaF1 I'm so sorry for the delayed response.
I checked the whole sections. The comments from my side are modified, and it looks good to me.

@HarshaF1 It seems the work is in progress [here](https://github.com/oopt-goldstone/goldstone-buildimage/pull/34/commits/3a0fb9a5d5420eb6609e91a023a9d58e0dae6154#diff-dbde925f7ca2be673c2dd59262f1adcb971ac33d90d3b1a90e118afd8d225ac5R386).
Do you plan to add more detailed explanation? Or, can we merge this PR?

HarshaF1 commented:
@KazuyaAnazawa we can merge this PR. This will be a place holder and good start for others to update the wiki content as required.  Once this PR is merged I will copy the content to the wiki page [here](https://github.com/oopt-goldstone/goldstone-buildimage/wiki)

KazuyaAnazawa commented:
@HarshaF1
Thank you for telling. I see, sorry for my misunderstanding.

@ishidawataru
We can merge the PR now. If it's ok, could you merge?

ishidawataru reviewed (changes requested):
@HarshaF1 Could you change the filename (wiki_review.md) to `README.md` ?

I'm not sure if it is a good idea to keep using the WiKi page because it doesn't have a review mechanism. How about just using the `docs` directory to store the documentation?

HarshaF1 commented:
@ishidawataru yes sure I will rename the file. Shall I keep this README.md in `docs` folder or replace the README.md file from the main folder ?

ishidawataru commented:
Let's keep the current README.md as is and keep the new file in `docs` directory.

ishidawataru commented:
@HarshaF1 Sorry for the delay. Merged the PR.
